packages/app-store/_utils/oauth/getTokenObjectFromCredential.ts

```diff
@@ -1,11 +1,26 @@
 import logger from "@calcom/lib/logger";
 import { safeStringify } from "@calcom/lib/safeStringify";
-import type { CredentialPayload } from "@calcom/types/Credential";
+import type {
+  CredentialForCalendarServiceWithEmail,
+  CredentialForCalendarServiceWithTenantId,
+  CredentialPayload,
+} from "@calcom/types/Credential";
 
 import { OAuth2TokenResponseInDbSchema } from "./universalSchema";
 
-export function getTokenObjectFromCredential(credential: CredentialPayload) {
-  const parsedTokenResponse = OAuth2TokenResponseInDbSchema.safeParse(credential.key);
+export function getTokenObjectFromCredential(
+  credential: CredentialPayload & {
+    delegatedTo?:
+      | CredentialForCalendarServiceWithEmail["delegatedTo"]
+      | CredentialForCalendarServiceWithTenantId["delegatedTo"];
+  }
+) {
+  let parsedTokenResponse;
+  if (credential?.delegatedTo?.key) {
```

Comment:
```suggestion
  if (credential.delegatedTo?.key) {
```

credential seems to be non nullable.